docs/essentials/exceptions/exceptions-hiearchy.md

```diff
@@ -0,0 +1,29 @@
+
+```
+Object
+  Exception
+    SystemException
+       InvalidCastException
+```
+
+Severe exceptions, such as [OutOfMemoryException](http://dotnet.github.io/api/System.OutOfMemoryException.html), are thrown by the runtime or in unrecoverable conditions.
+
+## Hierarchy of Runtime Exceptions
+
+The runtime has a base set of exceptions deriving from **Exception** that it throws when executing individual instructions. The following table hierarchically lists the standard exceptions provided by the runtime and the conditions under which you should create a derived class.
+
+| Exception type | Base type | Description | Example |
+| -------------- | --------- | ----------- | ------- |
+| [Exception](http://dotnet.github.io/api/System.Exception.html) | [Object](http://dotnet.github.io/api/System.Object.html) | Base class for all exceptions. | None (use a derived class of this exception). |
+| [IndexOutOfRangeException](http://dotnet.github.io/api/System.IndexOutOfRangeException.html) | [Exception](http://dotnet.github.io/api/System.Exception.html) | Thrown by the runtime only when an array is indexed improperly. | Indexing an array outside its valid range: `arr[arr.Length+1]` |
+| [NullReferenceException](http://dotnet.github.io/api/System.NullReferenceException.html) | [Exception](http://dotnet.github.io/api/System.Exception.html) | Thrown by the runtime only when a null object is referenced. | `object o = null; o.ToString();` |
+| [InvalidOperationException](http://dotnet.github.io/api/System.InvalidOperationException.html) | [Exception](http://dotnet.github.io/api/System.Exception.html) | Thrown by methods when in an invalid state. | Calling `Enumerator.GetNext()` after removing an Item from the underlying collection. |
+| [ArgumentException](http://dotnet.github.io/api/System.ArgumentException.html) | [Exception](http://dotnet.github.io/api/System.Exception.html) | Base class for all argument exceptions. | None (use a derived class of this exception). |
+| [ArgumentNullException](http://dotnet.github.io/api/System.ArgumentNullException.html) | [Exception](http://dotnet.github.io/api/System.Exception.html) | Thrown by methods that do not allow an argument to be null. | `String s = null; "Calculate".IndexOf (s);` |
+| [ArgumentOutOfRangeException](http://dotnet.github.io/api/System.ArgumentOutOfRangeException.html) | [Exception](http://dotnet.github.io/api/System.Exception.html) | Thrown by methods that verify that arguments are in a given range. | `String s = "string"; s.Chars[9];` |
+| [SEHException](http://dotnet.github.io/api/System.Runtime.InteropServices.SEHException.html) | [Exception](http://dotnet.github.io/api/System.Exception.html) | Exception encapsulating Win32 structured exception handling information. | Used in unmanaged code interop. |
```

Comment:
@kempb @mairaw `SEHException` is a Windows-only concept for COM interop.  It's a point-in-time concept that will eventually be replaced with a new P/Invoke contract.  I think it's probably best to leave it out so that we don't have to retroactively remove it, especially given that we don't want people relying too much on this stuff for new code.